azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/ApplicationGatewayImpl.java

```diff
@@ -1235,4 +1237,28 @@ public ApplicationGatewayTier tier() {
     public Update withoutPublicIPAddress() {
         return this.withoutPublicFrontend();
     }
+
+    // Actions
+
+    @Override
+    public void start() {
+        this.startAsync().await();
+        this.refresh();
+    }
+
+    @Override
+    public void stop() {
+        this.stopAsync().await();
+        this.refresh();
+    }
+
+    @Override
+    public Completable startAsync() {
+        return this.manager().inner().applicationGateways().startAsync(this.resourceGroupName(), this.name()).toCompletable();
```

Comment:
you can call `refreshAsync()` from `startAsync()` and remove `refresh()` from `start()`

```java
@Override
public Completable startAsync() {
        this.manager().inner().applicationGateways()
                .startAsync(this.resourceGroupName(), this.name())
                .flatMap(new Func1<Void, Observable<ApplicationGateway>>() {
                    @Override
                    public Observable<ApplicationGateway> call(Void aVoid) {
                        return refreshAsync();
                    }
                }).toCompletable();
}
```

This way both sync and async will have same behavior.